src/App.tsx

```diff
@@ -1,25 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import Pagination from './components/Pagination/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPages] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handleChangePerPages = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPages(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const startElement = (currentPage - 1) * perPage + 1;
+  const endElement = Math.min(currentPage * perPage, 42);
+
+  const showItems = items.slice(startElement - 1, endElement).map(item => (
+    <li data-cy="item" key={item}>
+      {item}
+    </li>
+  ));
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startElement} - {endElement} of 42)
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
```

Comment:
The 'perPage' state is being controlled by a select element. It's a good practice to validate user input even if it's coming from a controlled component to ensure that the component behaves as expected.